### PR TITLE
✨ add Week6

### PR DESCRIPTION
**Pull request by LouisaStrunk02**
Changes:

1. add tournament.py to world-cup
2. add mario-less, mario-more, cash, credit and readability
3. add dna.py

### CODE REVIEW

File: week6/sentimental-mario-less/mario.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+for i in range(height):
+
+    for j in range(height):
+        
+        if i + j < height -1:
+            print(" ", end="")
+        else:
+             print("#", end="")
+    print()
```
Comment (stefanluth) [inline week6/sentimental-mario-less/mario.py:17]: Place inside `main()`

File: week6/sentimental-mario-less/mario.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+while height < 1 or height > 8:
+
+    try:
+         height = int(input("Height(between 1 and 8, inclusive): "))
+    except ValueError:
+        print("That is not an int!")
+
+for i in range(height):
+
+    for j in range(height):
```
Comment (stefanluth) [inline week6/sentimental-mario-less/mario.py:11]:
```suggestion
for i in range(height):
    for j in range(height):
```

File: week6/sentimental-mario-less/mario.py (hunk 1 above)
Comment (stefanluth) [reply week6/sentimental-mario-less/mario.py:17]: Other files as well

File: week6/sentimental-credit/credit.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+def main():
+    cardnumber = int(input("Cardnumber: "))
+    used_Cardnumber = cardnumber
```
[the remark this answers is not in the document]
Comment (stefanluth) [reply week6/sentimental-credit/credit.py:3]: :sad snake_case noises:

File: week6/sentimental-readability/readability.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+def count_letters(text):
+
+    number_of_letters = 0
+
+    for i in range(len(text)):
+
+        if text[i].isalpha():
+            number_of_letters += 1
+    
+    return number_of_letters
```
Comment (stefanluth) [inline week6/sentimental-readability/readability.py:28]:
```suggestion
    return len([letter for letter in text if letter.isalpha()])
```
You could try to look into [list comprehensions](https://www.w3schools.com/python/python_lists_comprehension.asp). They are fun!

File: week6/sentimental-readability/readability.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+def count_words(text):
+
+    number_of_words = 1
+
+    for i in range(len(text)):
+
+        if text[i] == ' ':
+            number_of_words += 1
+    
+    return number_of_words
```
Comment (stefanluth) [inline week6/sentimental-readability/readability.py:39]:
```suggestion
    return len(text.split())
```

[str.split()](https://www.w3schools.com/python/ref_string_split.asp) creates a list of strings based on the applied string (text), split at every whitespace. You can then count the length of the resulting list. :)

File: week6/sentimental-readability/readability.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+def count_sentences(text):
+
+    number_of_sentences = 0
+
+    for i in range(len(text)):
+
+        if text[i] == '.' or text[i] == '!' or text[i] == '?':
+            number_of_sentences += 1
+
+    return  number_of_sentences
```
Comment (stefanluth) [inline week6/sentimental-readability/readability.py:50]: Try to apply my comment above in `count_words` to this function. Tip: Replace the sentence separators (`. ! ?`) to something else before splitting.

File: week6/sentimental-readability/readability.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+main()
```
Comment (stefanluth) [inline week6/sentimental-readability/readability.py:52]: `if __name__ =...`

File: week6/world-cup/tournament.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        sys.exit("Usage: python tournament.py FILENAME")
+
+    teams = []
+
+    with open(sys.argv[1]) as file:
+        reader = csv.DictReader(file)
+
+        for team in reader:
+            team["rating"] = int(team["rating"])
+            teams.append(team)
```
Comment (stefanluth) [inline week6/world-cup/tournament.py:19]:
```suggestion
    for team in reader:
        team["rating"] = int(team["rating"])
        teams.append(team)
```

File: week6/sentimental-readability/readability.py (hunk 6 above)
Comment (stefanluth) [reply week6/sentimental-readability/readability.py:50]: Tip2: Split takes an argument :)

File: week6/dna/dna.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        print("Usage: python3 dna.py *.csv *.txt")
+        exit()
+
+    database = []
+
+    with open (argv[1], "r") as file:
+        reader = csv.DictReader(file)
+
+        for name in reader:
+            database.append(name)
```
[the remark this answers is not in the document]
Comment (LouisaStrunk02) [reply week6/dna/dna.py:17]: Then I get the error: I/O operation on closed file.

File: week6/world-cup/tournament.py (hunk 8 above)
Comment (LouisaStrunk02) [reply week6/world-cup/tournament.py:19]: Same as above

File: week6/sentimental-readability/readability.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+    return len([letter for letter in text if letter.isalpha()])
+
+def count_words(text: str) -> int:
+
+    return len(text.split())
+
+def count_sentences(text: str) -> int:
+
+    sentences = len(text.split(". " or "! " or "? "))
```
Comment (stefanluth) [inline week6/sentimental-readability/readability.py:29]:
```suggestion
    sentences = len(text.split(". " or "! " or "? "))
```
This doesn't work. "or" is a boolean operator :)

**Review by MKrips — APPROVED** [review]

File: week6/sentimental-credit/credit.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        
+        evenposition = not evenposition
+        used_Cardnumber = used_Cardnumber // 10
+
+    if sum % 10 == 0:
+        is_AmericanExpress = length == 15 and (cardnumber // (10**13) == 34 or  cardnumber // (10**13) == 37)
+        is_MasterCard = length == 16 and (cardnumber // (10**14) == 51 or  cardnumber // (10**14) == 52 or cardnumber // (10**14) == 53 or cardnumber // (10**14) == 54 or cardnumber // (10**14) == 55)
+        is_VISA = (length == 16 and (cardnumber // (10**15) == 4)) or (length == 13 and (cardnumber // (10**12) == 4))
+
+        if is_AmericanExpress:
```
[the remark this answers is not in the document]
Comment (shoreal) [reply week6/sentimental-credit/credit.py:42]: I would say AmericanExpress is ok.

**Review by shoreal — APPROVED** [review]
What @stefanluth says, rest looks good :)

**Review by stefanluth — APPROVED** [review]

**Pull request merged**